### PR TITLE
Update data and concurrency distinctive lang feature descriptions

### PR DESCRIPTION
**Pull request by pcnfernando**
## Check List

- [ ] **Page Addition**
  - [ ] Add `permalink` to pages
  - [ ] If contains empty folder(s), Add front-matter `redirect_to:`

- [ ] **Page Rename**
  - [ ] Add front-matter `redirect_from`
  - [ ] Add front-matter `redirect_to:` (If applicable)

### CODE REVIEW

File: learn/guides/distinctive-language-features/concurrency.md [hunk 1]
```diff
@@ -413,7 +413,7 @@ transactional function bar() {
 }
 ```
 
-In the above code example, the named worker **``A``** has the ``transactional`` qualifier. Therefore the strand for execution of the code within **``A``** will have a new transaction that is branched from the calling transactional context within the function **``exec()``**.
+In the above code example, the named worker **``A``** has the ``transactional`` qualifier. Therefore, the strand for execution of the code within **``A``** will have a new transaction that is branched from calling transactional context within the function **``exec()``**.
```
Comment (MaryamZi) [inline learn/guides/distinctive-language-features/concurrency.md:416]: Why are we removing "the" here? I believe we need it.

```suggestion
In the above code example, the named worker **``A``** has the ``transactional`` qualifier. Therefore, the strand for execution of the code within **``A``** will have a new transaction that is branched from the calling transactional context within the function **``exec()``**.
```

File: learn/guides/distinctive-language-features/concurrency.md [hunk 2]
```diff
@@ -433,7 +433,7 @@ In the above code example, the function **``update()``** has a transactional con
 
 This is particularly useful when the **``update()``** function is called from a remote or resource transactional method, in a service object, and is invoked by another remote Ballerina program as a result of a service invocation. The transaction manager of the Ballerina program that initiated the transaction, will send a message to the Ballerina program in which this function is running. The two transaction managers follow a two-phase commit, such that when the remote Ballerina program knows that commit is successful, it will then arrange for the transaction manager of the Ballerina program running the **``update()``** function to call the commit handler.
 
-## Concurency Safety
+## Concurrency Safety
```
Comment (MaryamZi) [inline learn/guides/distinctive-language-features/concurrency.md:436]: Fixed in https://github.com/ballerina-platform/ballerina-dev-website/pull/3736/files#diff-3703f09cba3dd284401f066731bbec18a5d6cd760e6a609203e0e7dd50922306R436.

File: learn/guides/distinctive-language-features/concurrency.md [hunk 3]
```diff
@@ -506,7 +506,7 @@ table<Row> key(k) t = table [
 ];
 ```
 
-In the above code example, the string array **``s``** is declared as read-only using the type intersection operator ``&``. The type of **``s``** is both an array of strings as well as readonly, which means that it is an immutable array. This is enforced at compile-time to ensure that the values of the array **``s``** are immutable. Therefore it is safe to pass it as an argument to an ``isolated`` function.
+In the above code example, the string array **``s``** is declared as read-only using the type intersection operator ``&``. The type of **``s``** is both an array of strings and readonly, which means that it is an immutable array. This is enforced at compile-time to ensure that the values of the array **``s``** are immutable. Therefore, it is safe to pass it as an argument to an ``isolated`` function.
```
Comment (MaryamZi) [inline learn/guides/distinctive-language-features/concurrency.md:509]:
```suggestion
In the above code example, the string array **``s``** is declared as read-only using the type intersection operator ``&``. The type of **``s``** is both an array of strings and ``readonly,`` which means that it is an immutable array. This is enforced at compile-time to ensure that the values of the array **``s``** are immutable. Therefore, it is safe to pass it as an argument to an ``isolated`` function.
```

File: learn/guides/distinctive-language-features/data.md [hunk 4]
```diff
@@ -980,7 +978,7 @@ To access every element in the children of ``e``, for every element ``e`` in ``x
 
 To access every text item in the children of ``e``, for every element ``e`` in ``x``, use **`x/*.text()`**.
 
-To access every element named ``para`` in the descendants of ``e``, for every element ``e`` in ``x``, use **``x/**/<para>``**. Here the use of ``**`` signifies any number of levels within an xml element.
+To access every element named ``para`` in the descendants of ``e``, for every element ``e`` in ``x``, use **``x/**/<para>``**. Here the use of ``**`` signifies any number of levels within a xml element.
```
Comment (MaryamZi) [inline learn/guides/distinctive-language-features/data.md:981]:
```suggestion
To access every element named ``para`` in the descendants of ``e``, for every element ``e`` in ``x``, use **``x/**/<para>``**. Here the use of ``**`` signifies any number of levels within a XML element.
```

File: learn/guides/distinctive-language-features/concurrency.md (hunk 1 above)
Comment (pcnfernando) [reply learn/guides/distinctive-language-features/concurrency.md:416]: calling is used as a verb; isn't it?

File: learn/guides/distinctive-language-features/concurrency.md (hunk 2 above)
Comment (pcnfernando) [reply learn/guides/distinctive-language-features/concurrency.md:436]: Should be fine to have the commit here as well.

File: learn/guides/distinctive-language-features/concurrency.md (hunk 1 above)
Comment (MaryamZi) [reply learn/guides/distinctive-language-features/concurrency.md:416]: No, right? It's "calling transactional context" to mean the transactional context from which it is called?

File: learn/guides/distinctive-language-features/data.md [hunk 5]
```diff
@@ -980,7 +978,7 @@ To access every element in the children of ``e``, for every element ``e`` in ``x
 
 To access every text item in the children of ``e``, for every element ``e`` in ``x``, use **`x/*.text()`**.
 
-To access every element named ``para`` in the descendants of ``e``, for every element ``e`` in ``x``, use **``x/**/<para>``**. Here the use of ``**`` signifies any number of levels within an xml element.
+To access every element named ``para`` in the descendants of ``e``, for every element ``e`` in ``x``, use **``x/**/<para>``**. Here the use of ``**`` signifies any number of levels within a XML element.
```
Comment (MaryamZi) [inline learn/guides/distinctive-language-features/data.md:981]: Given that this is an abbreviation and the first letter is `X`, this should actually use "an" right?

```suggestion
To access every element named ``para`` in the descendants of ``e``, for every element ``e`` in ``x``, use **``x/**/<para>``**. Here the use of ``**`` signifies any number of levels within an XML element.
```

**Review by MaryamZi — APPROVED** [review]

**Pull request merged**